livre/lib/cache/HistogramObject.cpp

```diff
@@ -93,8 +93,27 @@ void binData( const SRC_TYPE* rawData,
                                          compCount * j * dataBlockSize.z() +
                                          compCount * k + c;
 
-                    const size_t binIndex = ( rawData[ index ] - histogram.getMin( )) /
-                                            range * ( binCount - 1 );
+                    size_t binIndex = 0;
+                    if( std::is_integral< SRC_TYPE >::value )
+                    {
+                        const size_t range =
+                                std::lround( histogram.getMax() - histogram.getMin( )) + 1u;
```

Comment:
I'd try to move the range calculation outside of the quadruple loop. You can use some template tricks to solve the problem of the different datatypes.